src/webots/gui/WbMainWindow.cpp

```diff
@@ -1822,12 +1775,26 @@ void WbMainWindow::showDocument(const QString &url) {
 }
 
 void WbMainWindow::showOnlineBook(const QString &book) {
-  QString versionString = WbApplicationInfo::version().toString();
-  versionString.replace(" revision ", "-rev");
+  QString versionString = WbApplicationInfo::branch();
+  if (versionString.isEmpty()) {
+    versionString = WbApplicationInfo::version().toString();
+    versionString.replace(" revision ", "-rev");
+  }
   const QString url = WbStandardPaths::cyberboticsUrl() + "/doc/" + book + "/index?version=" + versionString;
   showDocument(url);
 }
 
+void WbMainWindow::showOnlineDocumentationPage(const QString &book, const QString &page) {
```

Comment:
Since this method has a lot in common with `showOnlineBook`, I would create a common method called `showOnlineDocumentation` replacing these two methods, which takes a `book` argument and an optional `page` argument (empty by default).